### DISCUSSION TITLE
what will happen if enable both TCP and UDP tansport on publisher and subscriber side for one topic.

### DISCUSSION THREAD

**Discussion by hehenihao** [eProsima/Fast-DDS#2839] [Q&A]
Now i have a publisher and subscriber can communnicate using TCP transport  or UDP transport separately. I wonder which kind of tanport is preferred or both tranports will be in use if I enable both TCP and UDP tansports on publisher and subscriber side. 
I test this, but i'm confused by the reslut.

pushliser:
![image](https://user-images.githubusercontent.com/7553566/178921640-2701f611-19a9-4598-be83-e72cf5b9368b.png)
subscriber:
![image](https://user-images.githubusercontent.com/7553566/178922938-c01ac9c7-032f-4749-bae9-ef8251910032.png)

so the subscriber can receive data correctly.

but on wireshark, I get the both UPD and tcp tanffic. I can enusre that there is only one subscriber in the dds domain.

![image](https://user-images.githubusercontent.com/7553566/178922545-a6419d4d-c2e9-4396-8264-576ef4851277.png)

Can someone explain to me how can this happend?

**Comment by JLBuenoLopez** [comment]
Hi @hehenihao,

This is the expected behavior. Enabling both transports layers ensures that the sample is sent by TCP and UDP. The sample will be added to the DataReader History once it is received in any of the enabled transports. The second time the same sample is received is discarded because the DataReader already knows it. It is a redundancy mechanism.

**Reply by hehenihao** [reply]
Thanks for your quick reply, this mechanism can help in my use case.
